github_copilot_seats/seats.py

```diff
@@ -0,0 +1,122 @@
+import json
+import os
+from datetime import datetime
+
+import autokitteh
+from autokitteh.github import github_client
+from autokitteh.slack import slack_client
+from helpers import github_username_to_slack_user_id
+
+GITHUB_ORG = os.getenv("GITHUB_ORG")
+IDLE_USAGE_THRESHOLD = int(os.getenv("IDLE_USAGE_THRESHOLD"))
+LOGINS = os.getenv("LOGINS")
+LOG_CHANNEL = os.getenv("LOG_CHANNEL")
+
+logins = LOGINS.split(",")
+github = github_client("github_conn")
+slack = slack_client("slack_conn")
+
+
+def prune_idle_seats() -> list:
+    """Prunes idle GitHub Copilot users based on their last activity time."""
+    seats = find_idle_seats()
+    new_idle_seats = []
+    for seat in seats:
+        new_idle_seats.append(seat)
+        autokitteh.start(loc="seats.py:engage_seat", data={"event": {"seat": seat}})
+    return new_idle_seats
+
+
+def find_idle_seats() -> list:
+    """Identifies idle GitHub Copilot users based on their last activity time."""
+    seats = _get_all_seats()
+    t = datetime.now()
+    idle_seats = []
+
+    for seat in seats:
+        assignee_login = seat["assignee"]["login"]
+        last_activity_time = seat["last_activity_at"]
+
+        if logins and assignee_login not in logins:
+            print(f"skipping {assignee_login}")
+            continue
+
+        delta = t - last_activity_time
+        is_idle = delta >= IDLE_USAGE_THRESHOLD
+
+        print(
+            f"{assignee_login}: {t} - {last_activity_time} = {delta} "
+            f"{'>=' if is_idle else '<'} {IDLE_USAGE_THRESHOLD}"
+        )
+
+        if is_idle:
+            idle_seats.append(seat)
+
+    return idle_seats
+
+
+def engage_seat(seat: dict) -> None:
+    """Engages a GitHub user assigned to a seat by identifying their corresponding Slack user and initiating a workflow.
```

Comment:
specify in the log that this is meant to be run as a child workflow/session. (right?)